pkg/store/cache/caching_bucket_factory.go

```diff
@@ -104,10 +114,9 @@ func NewCachingBucketFromYaml(yamlContent []byte, bucket objstore.Bucket, logger
 	case string(GroupcacheBucketCacheProvider):
 		const basePath = "/_galaxycache/"
 
-		c, err = cache.NewGroupcache(logger, reg, backendConfig, basePath, r, bucket,
-			isTSDBChunkFile, isMetaFile, isBlocksRootDir,
-			config.MetafileExistsTTL, config.MetafileDoesntExistTTL, config.MetafileContentTTL,
-			config.ChunkObjectAttrsTTL, config.ChunkSubrangeTTL, config.BlocksIterTTL)
+		cfg.SetCacheImplementation(c)
```

Comment:
I don't think we need this line because `c` is still `nil` at this point, right?

Reply:
Yes.